api/src/main/java/org/openmrs/module/fhir/api/util/FHIRConditionUtil.java

```diff
@@ -69,7 +74,53 @@
 		return fhirCondition;
 	}
 
-	public static org.openmrs.Condition generateOpenMRSCondition() {
-		return null;
+	public static org.openmrs.Condition generateOpenMRSCondition(org.hl7.fhir.dstu3.model.Condition condition) {
+		org.openmrs.Condition openMrsCondition;
+
+		openMrsCondition = Context.getService(ConditionService.class).getConditionByUuid(condition.getId());
+		if (openMrsCondition == null) {
+			// No condition found to be updated
+			openMrsCondition = new org.openmrs.Condition();
+		}
+		BaseOpenMRSDataUtil.readBaseExtensionFields(openMrsCondition, condition);
+
+		openMrsCondition.setId(Integer.parseInt(condition.getId()));
+
+		// set patient
+		if (condition.getSubject() != null) {
+			String patientRef = condition.getSubject().getReference();
+			Patient openMrsPatient = Context.getPatientService().getPatientByUuid(patientRef);
+			openMrsCondition.setPatient(openMrsPatient);
+		}
+
+		// set condition Uuid
+		openMrsCondition.setUuid(condition.getContext().getReference());
+		try {
+			openMrsCondition.setOnsetDate(condition.getOnsetDateTimeType().getValue());
+		} catch (FHIRException e) {
+			e.printStackTrace();
+		}
+		openMrsCondition.setAdditionalDetail(condition.getNoteFirstRep().getText());
+		openMrsCondition.setDateChanged(condition.getAssertedDate());
+
+		// set concept
+		if (condition.getCode() != null) {
+			CodeableConcept conceptDt = condition.getCode();
+			List<Coding> codings = conceptDt.getCoding();
+
+			//Set concept
+			Concept openMrsConcept = new Concept();
```

Comment:
As discussed on slack, the request to create a new condition should throw an exception if the concept does not already exist in the openmrs concept dictionary. We should not be creating new concepts here.

Reply:
Noted